python/dolfinx/io/gmsh.py

```diff
@@ -0,0 +1,120 @@
+# Copyright (C) 2022 Jørgen S. Dokken
+#
+# This file is part of DOLFINx (https://www.fenicsproject.org)
+#
+# SPDX-License-Identifier:    LGPL-3.0-or-later
+"""Tools to extract data from GMSH models"""
+import typing
+
+import numpy as np
+
+import ufl
+from dolfinx import cpp as _cpp
+from dolfinx.mesh import CellType
+
+__all__ = ["extract_topology_and_markers", "extract_geometry", "ufl_mesh", "cell_perm"]
+
+
+def extract_topology_and_markers(gmsh_model, model_name=None):
+    """Extract all entities tagged with a physical marker in the gmsh
```

Comment:
Add `Arguments` and `Returns` field to the doc string. Can typing be introduced without importing gmsh?